jsonschema/src/output.rs

```diff
@@ -169,11 +231,56 @@ impl<T> OutputUnit<T> {
             value: error,
         }
     }
+
+    ///  The location in the schema of the keyword
+    pub const fn keyword_location(&self) -> &JSONPointer {
+        &self.keyword_location
+    }
+
+    ///  The absolute location in the schema of the keyword. This will be
+    ///  different to `keyword_location` if the schema is a resovled reference.
+    pub const fn absolute_keyword_location(&self) -> &Option<AbsolutePath> {
+        &self.absolute_keyword_location
+    }
+
+    ///  The location in the instance
+    pub const fn instance_location(&self) -> &JSONPointer {
+        &self.instance_location
+    }
 }
 
+impl OutputUnit<Annotations<'_>> {
+    /// The annotations found at this output unit
+    pub fn value(&self) -> Cow<'_, serde_json::Value> {
+        self.value.value()
+    }
+}
+
+impl OutputUnit<ErrorDescription> {
+    /// The error for this output unit
+    pub const fn error_description(&self) -> &ErrorDescription {
+        &self.value
+    }
+}
+
+/// Annotations associated with an output unit.
 #[derive(serde::Serialize, Debug, Clone, PartialEq)]
 pub struct Annotations<'a>(AnnotationsInner<'a>);
 
+impl<'a> Annotations<'a> {
+    /// The `serde_json::Value` of the annotation
+    pub fn value(&'a self) -> Cow<'a, serde_json::Value> {
+        match &self.0 {
+            AnnotationsInner::Value(v) => Cow::Borrowed(v),
+            AnnotationsInner::ValueRef(v) => Cow::Borrowed(v),
+            AnnotationsInner::UnmatchedKeywords(kvs) => {
+                let value = serde_json::to_value(kvs).unwrap();
```

Comment:
Could we use `expect` here instead? Not sure what would be the situation when panic may happen

Reply:
I've added an `expect`. The error cases mentioned in `serde_json::to_value` are:

> This conversion can fail if `T`'s implementation of `Serialize` decides to fail, or if `T` contains a map with non-string keys.

But `T` in  our case is `AHashMap<String, serde_json::Value>`, which does not fall into either of these cases.

Reply:
Thanks!